sunpy/util/util.py

```diff
@@ -92,12 +92,14 @@ def expand_list(inp):
     ----------
     * https://stackoverflow.com/questions/2185822/expanding-elements-in-a-list/2185971#2185971
     """
-    return [item for item in expand_list_generator(inp)]
+    return list(expand_list_generator(inp))
 
 
 def expand_list_generator(inp):
+    exclude_union = (str | bytes)
+    include_union = (list | tuple | Iterator)
     for item in inp:
-        if isinstance(item, tuple | list | UserList):
+        if not isinstance(item, exclude_union) and isinstance(item, include_union):
```

Comment:
```suggestion
        if isinstance(item, include_union):
```
Would this be enough?

Reply:
I don't think so, what if someone passes a byte string?

Reply:
Would that not be caught by this check? 

Might be worth adding some unit tests to check `expand_list_generator` behavior?

Reply:
We are checking for the list which in turn checks for the generator as input which solves the problem but here I'm adding an extra check for byte string

Reply:
But why would a byte string hit either of those categories?

Reply:
> While I doubt someone will pass a byte string in the context of this MR, as this is a utility function, I believe you want to avoid an issue with byte strings. In the current implementation a byte string wouldn't be caught by `isinstance(item, str)` as a regular string or raw string literal would be.

As suggested by @PaulJWright

Reply:
Sure, but I mean why would a byte string now trigger this line of code?

Reply:
It triggers exclude_union which in turn triggers a string or byte

Reply:
I guess we don't want any unit tests for byte strings

Reply:
But in this suggestion, we only check for include_union which is list, tuple and Iterator. 

I would not have assumed a byte string would trigger the last one but I don't know what Iterator fully covers.

Reply:
Why would we not want a unit test for it?

Reply:
We should expand the unit test to see what happens if we use a numpy array, a fits header, a astropy WCS, a byte string and maybe some others?

Reply:
Yeah, this suggestion doesn't covers exclude_union.

Reply:
> Why would we not want a unit test for byte strings?

If we have to remove byte dependency then there is no need of byte test cases

Reply:
> Yeah, this suggestion doesn't covers exclude_union.

Yeah, since we I don't we would need it.

Reply:
> > Why would we not want a unit test for byte strings?
> 
> If we have to remove byte dependency then there is no need of byte test cases

I am not sure I follow?

Reply:
Ah! Never mind. I'll add unit tests for the data types and objects mentioned above

Reply:
I updated the test and the code, see what you think.

Reply:
Seems good. What about Astropy WCS object and numpy, should I add them?

Reply:
Those should be covered by the .data and .wcs from the sunpy map.